bin/proving-service/src/api/prover.rs

```diff
@@ -0,0 +1,188 @@
+use miden_block_prover::LocalBlockProver;
+use miden_objects::{
+    MIN_PROOF_SECURITY_LEVEL, batch::ProposedBatch, block::ProposedBlock,
+    transaction::TransactionWitness, utils::Serializable,
+};
+use miden_tx::{LocalTransactionProver, TransactionProver};
+use miden_tx_batch_prover::LocalBatchProver;
+use tokio::sync::Mutex;
+use tonic::{Request, Response, Status};
+use tracing::instrument;
+
+use crate::{
+    commands::worker::ProverType,
+    generated::{ProofType, ProvingRequest, ProvingResponse, api_server::Api as ProverApi},
+    utils::MIDEN_PROVING_SERVICE,
+};
+
+/// The provers for the proving service.
+///
+/// This enum is used to store the provers for the proving service.
+/// Only one prover is enabled at a time.
+enum Provers {
```

Comment:
nit: This could be changed to singular (`Prover`) or something `RequestProver` maybe